ejector/ejection_transactor.go

```diff
@@ -22,3 +31,257 @@ type EjectionTransactor interface {
 	// Complete the ejection proceedings against the operator with the given address.
 	CompleteEjection(ctx context.Context, addressToEject gethcommon.Address) error
 }
+
+var _ EjectionTransactor = &ejectionTransactor{}
+
+// ejectionTransactor is the production implementation of the EjectionTransactor interface.
+//
+// The ejection transactor is thread safe, although parallel calls may result in duplicate work (i.e. two calls might
+// end up putting the same data in a cache).
+type ejectionTransactor struct {
+	logger logging.Logger
+
+	// The address of this ejector instance.
+	selfAddress gethcommon.Address
+
+	// Used to execute eth reads
+	caller *contractEigenDAEjectionManager.ContractIEigenDAEjectionManagerCaller
+
+	// Used to execute eth writes
+	transactor *contractEigenDAEjectionManager.ContractIEigenDAEjectionManagerTransactor
+
+	// A function that can sign transactions from selfAddress.
+	signer bind.SignerFn
+
+	// A utility for getting the reference block number.
+	referenceBlockProvider eth.ReferenceBlockProvider
+
+	// A utility for looking up which quorums a given validator is a member of at a specific reference block number.
+	validatorQuorumLookup eth.ValidatorQuorumLookup
+
+	// A utility for converting between validator IDs and addresses.
+	validatorIDToAddressConverter eth.ValidatorIDToAddressConverter
+
+	// The maximum gas limit to use for transactions.
+	maxGasOverride uint64
+}
+
+// Create a new EjectionTransactor.
+func NewEjectionTransactor(
+	logger logging.Logger,
+	client bind.ContractBackend,
+	ejectionContractAddress gethcommon.Address,
```

Comment:
Should you also validate that these aren't 0x0 contract addresses?

Reply:
Added checks. I don't mind defensive checks like this, but I don't always think they are necessary. Specifically for contract addresses, the offchain contract directory code will return an error if the address is a zero address.